src/Linux/Avalonia.LinuxFramebuffer/Output/DrmBindings.cs

```diff
@@ -144,31 +145,43 @@ public unsafe class DrmCard : IDisposable
         public int Fd { get; private set; }
         public DrmCard(string path = null)
         {
-            if(path == null)
-            {
-                var files = Directory.GetFiles("/dev/dri/");
-
-                foreach(var file in files) 
-                {
-                    var match = Regex.Match(file, "card[0-9]+");
-
-                    if(match.Success)
-                    {
-                        Fd = open(file, 2, 0);
-                        if(Fd != -1) break; 
-                    }    
-                }
+            path = path ?? "/dev/dri/card0";
+            Fd = open(path, 2, 0);
+            if (Fd == -1)
+                throw new Win32Exception("Couldn't open " + path);
+        }
 
-                if(Fd == -1) throw new Win32Exception("Couldn't open /dev/dri/card[0-9]+");
+        public static bool TryCreate(string path, [CanBeNull] out DrmCard drmCard)
+        {
+            try
+            {
+                var card = new DrmCard(path);
+                drmCard = card;
+                return true;
             }
-            else 
+            catch (Exception e)
```

Comment:
This swallows the exception, so there is nothing to report later if all outputs weren't available

Reply:
it's done on LinuxFramebufferPlattform L196.
Oh I see would be better that we check the file descriptor instead of catching the exception

Reply:
On AutoDetection an error will be thrown if no output is available

Reply:
I still don't get why we want to have that at all. We throw an exception in the constructor just to catch it here. I would not have the public constructor at all. I would only have a constructor that receives the fd.